Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.mm

```diff
@@ -455,6 +455,14 @@ static float toNSURLSessionTaskPriority(WebCore::ResourceLoadPriority priority)
 
     if (isTopLevelNavigation())
         request.setFirstPartyForCookies(request.url());
+    else {
+        WebCore::RegistrableDomain firstPartyDomain { request.firstPartyForCookies() };
+        if (auto* storageSession = m_session->networkStorageSession(); storageSession
```

Comment:
Please break out the two checks – redirect and request – into two local variables with good names. That will make this if statement much more readable.